onnx_tf/handlers/backend/gather.py

```diff
@@ -12,3 +12,22 @@ class Gather(BackendHandler):
   @classmethod
   def version_1(cls, node, **kwargs):
     return [cls.make_tensor_from_onnx_node(node, **kwargs)]
+
+  @classmethod
+  def version_11(cls, node, **kwargs):
+    x = kwargs["tensor_dict"][node.inputs[0]]
+    indices = kwargs["tensor_dict"][node.inputs[1]]
+    axis = node.attrs.get("axis", 0)
+    dimension = tf.cast(tf.shape(x)[axis], indices.dtype)
+    indices_shape = tf.shape(indices)
+    dim_broadcasted = tf.broadcast_to(dimension, indices_shape)
+    nonneg_indices = tf.cast(tf.greater_equal(indices, 0), dtype=indices.dtype)
+    negative_indices = tf.cast(tf.less(indices, 0), dtype=indices.dtype)
+    final_indices = dim_broadcasted * negative_indices + negative_indices * indices + nonneg_indices * indices
+    cond1 = tf.greater_equal(final_indices, 0)
+    cond2 = tf.less(final_indices, dimension)
+    indice_valiation = tf.reduce_all(tf.logical_and(cond1, cond2))
```

Comment:
@winnietsang for the `gather`, it is simpler than the `*_nd` ops and I try to avoid the `tf.while` invocation, so here I did the validation directly after computing the final indices.

Reply:
@winston-zillow Please extend the Gather handler with GatherAndScatterMixin.
GatherAndScatterMixin is a class to provide common functions to handle tasks needed in Gather and Scatter related handlers 

For example  to check are there any indices are out of bounds can simply call 
result = cls.chk_idx_out_of_bounds_along_axis(data, axis, indices)

For process negative indices can call the following common function
indices = cls.process_neg_idx_along_axis(data, axis, indices)

Reply:
@winnietsang done. didn't notice the `*along_axis` versions.